core-common/src/main/java/org/zalando/nakadi/repository/kafka/KafkaTopicRepository.java

```diff
@@ -306,31 +305,11 @@ public void syncPostBatch(
             throws EventPublishingException {
         final Producer<byte[], byte[]> producer = kafkaFactory.takeProducer();
         try {
-            final Map<String, String> partitionToBroker = producer.partitionsFor(topicId).stream()
-                    .filter(partitionInfo -> partitionInfo.leader() != null)
-                    .collect(
-                            Collectors.toMap(
-                                    p -> String.valueOf(p.partition()),
-                                    p -> p.leader().idString() + "_" + p.leader().host()));
-            batch.forEach(item -> {
-                Preconditions.checkNotNull(
```

Comment:
I think checking for partition is actually useful.  If we don't specify it, Kafka will do its own partitioning.

Reply:
yes, I cut too much..